docs/syntax_and_semantics/offsetof.md

```diff
@@ -1,9 +1,19 @@
 # offsetof
 
-An `offsetof` expression returns the byte offset of an instance variable in a struct or class type. It accepts the type as first argument and the instance variable name prefixed by an `@` as second argument:
+An `offsetof` expression returns the byte offset of a subobject in an instance of a class or struct.
+
+There are two forms of `offsetof` expressions. The first form accepts any type as first argument and an instance variable name prefixed by an `@` as second argument:
+
+```crystal
+# x : Type
+offsetof(Type, @ivar) # the byte offset of x.@ivar relative to x
+```
+
+The second form accepts any tuple type as first argument and an integer literal index as second argument:
```

Comment:
```suggestion
The second form accepts a tuple type as first argument and an integer literal index as second argument:
```
"any" could be interpeted to include named tuples, but this only works for (non-named) tuples.

Reply:
What about "`Tuple` type"? (with a link to `Tuple`'s API docs page)